src/Automation/SnapshotCommand.cs

```diff
@@ -35,28 +36,57 @@ public static ScanResults Execute(Config config, IScanTools scanTools)
             if (config.CustomUIAConfigPath != null)
                 scanTools.Actions.RegisterCustomUIAPropertiesFromConfig(config.CustomUIAConfigPath);
 
+            List<ScanResults> resultList = new List<ScanResults>();
+
             var rootElements = scanTools.TargetElementLocator.LocateRootElements(config.ProcessId);
 
-            return scanTools.Actions.Scan(rootElements?.First(), (element, elementId) =>
+            if (rootElements is null || !rootElements.Any())
+            {
+                return resultList;
+            }
+
```

Comment:
In the case where an app has multiple roots but doesn't specify the flag, we will fail silently, which seems non-ideal. I'm wondering if we should defend this more aggressively and throw an Exception if a multi-root app is scanned without the multi-app flag. That's technically a breaking change for anyone currently using the `IScanner` interface on a multi-root app, but I think we can defend it since multi-root apps are not common (and those that are using the current tool are failing to report all results). As a side note, that also removes the need for the early exit (lines 57-61)

Reply:
This is a tough one. On the one hand, I would argue we should break existing usage as little as possible, however notifying people that they are possibly missing out on accessibility issues is also a fair point. I'm a bit worried that for some folks, calling Scan might get a bit annoying if they have to encounter this exception; I would prefer having people use Scan like they used to regardless if there are multiple windows or not.

Reply:
I discussed this with @RobGallo, and let's keep the backward compatibility. As he put it, "They should know they have windows that aren't being scanned."